### DISCUSSION TITLE
How to create a password field?

### DISCUSSION THREAD

**Discussion by rairafaqatali** [elsa-workflows/elsa-core#1498] [Q&A]
Hi @sfmskywalker ,
I want to create a password field in my custom activity. 

Can you kindly let me know how I'll achieve that?
I tried with this snippet but got no luck.
`       
        [ActivityInput(Hint = "Authorization Token")]
        [DataType(DataType.Password)]
        public string AuthToken { get; set; }`

**Comment by sfmskywalker** [comment]
Sure. There are a few steps involved with creating your own field types / UI hints:

1. Define your own UI hint (which you did as `"Authorization Token", but if you want to be able to reuse the password field, it might be better to call it `"Password"` or `"Secret"` to indicate that it's a secret value).
2. Create a field type driver (with JavaScript).
3. Create a designer plugin that registers your custom field driver.

Read about custom plugins [here](https://elsa-workflows.github.io/elsa-core/docs/next/extensibility/extensibility-designer-plugins#custom-plugins) and custom field types [here](https://elsa-workflows.github.io/elsa-core/docs/next/extensibility/extensibility-designer-plugins#custom-field-types).

Alternatively (or additionally), you could create a new issue requesting the addition of a new `"Secret"` UI hint. And perhaps you might even submit a PR 😄
